eth2near/eth2near-block-relay-rs/src/eth2near_relay.rs

```diff
@@ -140,60 +149,130 @@ impl Eth2NearRelay {
                 );
 
                 if self.eth_client_contract.is_known_block(&hash)? {
-                    return Ok(true);
+                    trace!(target: "relay", "Block with slot={} was found on NEAR", slot);
+                    Ok(true)
                 } else {
                     trace!(target: "relay", "Block with slot={} not found on Near", slot);
-                    return Ok(false);
+                    Ok(false)
                 }
             }
             Err(err) => {
-                trace!(target: "relay", "Error \"{}\" in getting beacon block body for slot={}", err, slot)
+                trace!(target: "relay", "Error \"{}\" in getting beacon block body for slot={}", err, slot);
+                Err(err)?
             }
         }
+    }
+
+    // get the slot numbers between the last submitted slot and signature slot for next update
+    // if we sending updates once in 'update_submission_frequency' epochs
+    // `update_submission_frequency * ONE_EPOCH_IN_SLOTS` -- gap in slots between two finalized
+    //  blocks in neighboring updates.
+    // `2 * ONE_EPOCH_IN_SLOTS` -- gap between finalized and attested block.
+    // `1` -- expected gap between attested block slot and signature slot
+    fn get_gap_between_finalized_and_signature_slot(update_submission_frequency: u64) -> u64 {
+        update_submission_frequency * ONE_EPOCH_IN_SLOTS + 2 * ONE_EPOCH_IN_SLOTS + 1
```

Comment:
Still, instead of "2", we should have a constant with a self-explaining name, why it's actually "2"

Reply:
fixed: https://github.com/aurora-is-near/rainbow-bridge/pull/775/commits/cf032d3ed2d425e9c94c26929f1ae7f7c3dfbb50